src/tricicl/metrics/representation_shift.py

```diff
@@ -0,0 +1,247 @@
+        Resets the last representation.
+
+        This will preserve the initial representation value of each experience.
+        To be used at the beginning of each eval experience.
+
+        :return: None.
+        """
+        self.shifting.reset_last()
+
+    def update(self, k: int, v: Tensor, initial=False):
+        """
+        Update shifting metric.
+        See `RepresentationShift` for more detailed information.
+
+        :param k: key to update
+        :param v: value associated to k
+        :param initial: update initial value. If False, update
+            last value.
+        """
+        self.shifting.update(k, v, initial=initial)
+
+    def result(self, k=None) -> Union[None, float, Dict[int, float]]:
+        """
+        See `RepresentationShift` documentation for more detailed information.
+
+        k: optional key from which compute forgetting.
+        """
+        return self.shifting.result(k=k)
+
+    def before_training_exp(self, strategy: "BaseStrategy") -> None:
+        self.train_exp_id = strategy.experience.current_experience
+
+    def before_eval(self, strategy) -> None:
+        self.reset_last_accuracy()
+
+    def before_eval_exp(self, strategy: "BaseStrategy") -> None:
+        self._current_representation.reset()
+
+    def after_eval_iteration(self, strategy: "BaseStrategy") -> None:
+        super().after_eval_iteration(strategy)
+        self.eval_exp_id = strategy.experience.current_experience
+        self._current_representation.update(logits=strategy.logits)
+
+    def after_eval_exp(self, strategy: "BaseStrategy") -> MetricResult:
+        # update experience on which training just ended
+        if self.train_exp_id == self.eval_exp_id:
+            self.update(self.eval_exp_id, self._current_representation.result(), initial=True)
+        else:
+            # update other experiences
+            # if experience has not been encountered in training
+            # its value will not be considered in shifting
+            self.update(self.eval_exp_id, self._current_representation.result())
+
+        # this checks if the evaluation experience has been
+        # already encountered at training time
+        # before the last training.
+        # If not, shifting should not be returned.
+        if self.result(k=self.eval_exp_id) is not None:
+            return self._package_result(strategy)
+
+    def get_global_counter(self):
+        return self.x
+
+    def after_training(self, strategy: "BaseStrategy") -> "MetricResult":
+        self.x += 1
+
+    def _package_result(self, strategy: "BaseStrategy") -> MetricResult:
+
+        shifting = self.result(k=self.eval_exp_id)
+        metric_name = get_metric_name(self, strategy, add_experience=True)
+        plot_x_position = self.get_global_counter()
+
+        metric_values = [MetricValue(self, metric_name, shifting, plot_x_position)]
+        return metric_values
+
+    def __str__(self):
+        return f"Experience{self.shifting}"
```

Comment:
The metric names are quite long on the graphs, what about `CosineRepresentationShift` and `L2RepresentationShift`?